unit_test/fuzzing/test_responder/test_spdm_responder_chunk_send_ack/chunk_send_ack.c

```diff
@@ -84,64 +84,51 @@ void libspdm_test_responder_chunk_send_ack_case1(void **State)
-            spdm_message_header_t *test_spdm_message_header_t;
+
+            /* Remaining space should meet the chunk_size. */
+            if(bytes_total - sizeof(spdm_chunk_send_request_t) - sizeof(uint32_t) <
+               chunk_send_request->chunk_size) {
+                break;
+            }
 
             request_size = sizeof(spdm_chunk_send_request_t) + sizeof(uint32_t) +
                            chunk_send_request->chunk_size;
 
-            if (chunk_send_request->chunk_size > bytes_total) {
-                break;
-            }
-            if (chunk_send_request->chunk_size >
-                (request_size - sizeof(spdm_chunk_send_request_t) - sizeof(uint32_t)) ||
-                chunk_send_request->chunk_size == 0) {
-                break;
-            }
-            test_spdm_message_header_t =
-                (spdm_message_header_t *)(((uint8_t*) (chunk_send_request + 1)) + sizeof(uint32_t));
-            if (test_spdm_message_header_t->request_response_code == SPDM_RESPOND_IF_READY) {
-                break;
-            }
             spdm_context->local_context.capability.data_transfer_size =
                 chunk_send_request->chunk_size + sizeof(spdm_chunk_send_request_t) +
                 sizeof(uint32_t);
         } else {
-            chunk_send_request = (spdm_chunk_send_request_t *)request;
-            spdm_message_header_t *test_spdm_message_header_t;
-
-            request_size = sizeof(spdm_chunk_send_request_t) + chunk_send_request->chunk_size;
-
-            if (chunk_send_request->chunk_size > bytes_total) {
-                break;
-            }
-            if (chunk_send_request->chunk_size >
-                (request_size - sizeof(spdm_chunk_send_request_t) - bytes_sent) ||
-                chunk_send_request->chunk_size == 0) {
+            /* The remaining number of bytes should not be less than the minimum effective size*/
+            if(bytes_total - bytes_sent < sizeof(spdm_chunk_send_request_t)) {
                 break;
             }
+            chunk_send_request = (spdm_chunk_send_request_t *)request;
 
-            test_spdm_message_header_t =
-                (spdm_message_header_t *)(((uint8_t*) (chunk_send_request + 1)));
-            if (test_spdm_message_header_t->request_response_code == SPDM_RESPOND_IF_READY) {
+            /* Remaining space should meet the chunk_size. */
+            if(bytes_total - bytes_sent - sizeof(spdm_chunk_send_request_t) <
+               chunk_send_request->chunk_size) {
                 break;
             }
 
+            request_size = sizeof(spdm_chunk_send_request_t) + chunk_send_request->chunk_size;
+
             spdm_context->local_context.capability.data_transfer_size =
                 chunk_send_request->chunk_size + sizeof(spdm_chunk_send_request_t);
-
-            if (bytes_total - bytes_sent == chunk_send_request->chunk_size) {
-                chunk_send_request->header.param1 = SPDM_CHUNK_SEND_REQUEST_ATTRIBUTE_LAST_CHUNK;
```

Comment:
why change this?

Reply:
Thank you, I have made the modifications. I will commit this modification with a new PR.